src/base/utils/tar.cpp

```diff
@@ -0,0 +1,298 @@
+                return;
+
+            // Invalid header: The first header field represents the filename.
+            // If the 1st byte is null then we can't extract a name. So the whole header becomes invalid to us.
+            m_valid = !m_header.startsWith('\0');
+            verifyAndCheckNull();
+        }
+
+        QString name() const
+        {
+            if (!isValid())
+                return {};
+
+            const QByteArray filename = Utils::ByteArray::midView(m_header, TAR_FILENAME_OFFSET, TAR_FILENAME_SIZE);
+            const int filenameLen = strlen(filename);
+            if (filenameLen == 0)
+                return {};
+
+            const QByteArray filenamePrefix = Utils::ByteArray::midView(m_header, TAR_FILENAME_PREFIX_OFFSET, TAR_FILENAME_PREFIX_SIZE);
+            const int filenamePrefixLen = strlen(filenamePrefix);
+
+            const QString finalFilename = (filenamePrefixLen > 0) ?
+                                        QString::fromUtf8(filenamePrefix.constData(), filenamePrefixLen)
+                                              + '/' + QString::fromUtf8(filename.constData(), filenameLen)
+                                      : QString::fromUtf8(filename.constData(), filenameLen);
+
+            return Utils::Fs::toValidFileSystemName(finalFilename, true);
+        }
+
+        bool isNull() const { return m_null; }
+        bool isValid() const { return m_valid; }
+        bool isVerified() const { return m_verified; }
+        Types type() const { return isValid() ? static_cast<Types>(m_header.at(TAR_FILETYPE_OFFSET)) : INVALID; }
+        int size() const { return isValid() ? octalToDec(Utils::ByteArray::midView(m_header, TAR_FILESIZE_OFFSET, TAR_FILESIZE_SIZE)) : 0; }
+
+    private:
+        int octalToDec(const QByteArray &data) const
+        {
+            // only the ASCII characters from '0' to '7' are valid
+            int i = 0;
+            for (; i < data.size(); ++i) { // skip leading invalid bytes
+                char byte = data.at(i);
+                if (byte >= '0' && byte <= '7')
+                    break;
+            }
+
+            int fileSize = 0;
+            for (; i < data.size(); ++i) {
+                char byte = data.at(i);
+                if (byte < '0' || byte > '7') // skip trailing invalid bytes
+                    break;
+
+                fileSize = (8 * fileSize) + (byte - '0');
+            }
+
+            return fileSize;
+        }
+
+        // The various header fields that hold strings are of known max size.
+        // So in the case the string occupies their max size, the string will not
+        // be null terminated. It is pointless to have std::strlen() run through
+        // the whole buffer when we already know the upper limit. Also the buffer
+        // isn't guaranteed to contain a null byte and std::strlen() will have
+        // undefined behavior.
+        int strlen(const QByteArray &str) const
```

Comment:
IMO better differentiate the name with std::strlen() to avoid confusion.

Reply:
There is `uint qstrnlen(const char *str, uint maxlen)` that you can use instead (or at least as underlying one).